Add unit tests for NumberRangePreferenceCompat dialog behavior (#13283)

pankaj0695 opened the pull request:
<!--- Please fill the necessary details below -->
## Purpose / Description
This PR adds missing unit test coverage for NumberRangePreferenceCompat, specifically the behavior marked with @NeedsTest where the preference value should remain unchanged when the dialog is confirmed with blank input. It also adds coverage for range validation helpers to prevent regressions.

## Fixes
* Fixes somme of #13283

## Approach
- Added a new Robolectric test class at AnkiDroid/src/test/java/com/ichi2/preferences/NumberRangePreferenceCompatTest.kt.
- Implemented focused unit tests for:
  - getValidatedRangeFromInt: values within range and below min clamping.
  - getValidatedRangeFromString: empty string, valid numeric string and invalid string handling.
  - onDialogClosed: ensures blank input and cancelled dialog do not trigger value updates, while valid input does.

## How Has This Been Tested?
Run the unit tests via Gradle
```kotlin
./gradlew :AnkiDroid:testDebugUnitTest --tests "com.ichi2.preferences.NumberRangePreferenceCompatTest"
```
<img width="1325" height="324" alt="Screenshot 2025-12-28 at 6 31 34 PM" src="https://github.com/user-attachments/assets/e49d972c-b516-4826-80cc-3414eb19ffbf" />

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

<!--- Uncomment this section ONLY if this PR introduces new resources (external libraries, icons etc)
## Licenses
_For each new external resource, add a row to the table below:_

| Library | Description | License |
| --- | --- | --- |
| Sample Icon Library | Sample Description | [The Apache Software License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.txt) |

**Maintainers:**

* [ ] Add the https://github.com/ankidroid/Anki-Android/labels/Licenses label
* [ ] Update the [licenses](https://github.com/ankidroid/Anki-Android/wiki/Licences) wiki when merging
--->

david-allison commented on AnkiDroid/src/test/java/com/ichi2/preferences/NumberRangePreferenceCompatTest.kt:
```diff
@@ -0,0 +1,189 @@
+    fun setUp() {
+        context = ApplicationProvider.getApplicationContext()
+        preference = TestableNumberRangePreferenceCompat(context)
+    }
```
```kotlin
class NumberRangePreferenceCompatTest: RobolectricTest() {
    private val preference = TestableNumberRangePreferenceCompat(targetContext)
```

pankaj0695 commented:
Thanks for the review, I addressed the nitpicks by extracting the repeated dialog setup into a helper and switching the onDialogClosed assertions to MockK interaction verification (no functional changes). I also squashed the commits and updated the commit message.

pankaj0695 commented:
Thanks for the note, I did use LLM for guidance, but I made sure I understood the changes before pushing. I’ve now removed the unnecessary comments and made required changes, and will make sure future updates are clean and reviewed before I push them.

criticalAY commented on AnkiDroid/src/test/java/com/ichi2/preferences/NumberRangePreferenceCompatTest.kt:
```diff
@@ -0,0 +1,130 @@
+    private val preferenceMock: NumberRangePreferenceCompat = mockk(relaxed = true)
+
+    @Test
+    fun getValidatedRangeFromInt_withinRange_returnsInput() {
```
Use backticks or camel case

criticalAY reviewed (changes requested):
Looks good one last change

pankaj0695 commented:
I have made the required updates

david-allison reviewed (approved):
cheers!
